include/fastdds/rtps/security/cryptography/CryptoKeyFactory.h

```diff
@@ -138,29 +140,82 @@ class CryptoKeyFactory
      * @return TRUE is successful
      */
     virtual bool unregister_participant(
-            ParticipantCryptoHandle* participant_crypto_handle,
+            std::shared_ptr<ParticipantCryptoHandle>& participant_crypto_handle,
             SecurityException& exception) = 0;
 
     /**
-     * Releases resources associated with a DataWriter. The Crypto Handle becomes unusable after this
+     * Releases resources associated with a DataWriter. The Crypto Handle may become unusable after this
```

Comment:
This method clear the framework references to the writer keys.
That is, the participant key keeps `shared_ptrs` to its readers and writers keys. Calling this method the participant reset that ptr.
If some decoding is undergone while this method is called the handle is still valid because it will be clear when the last stack reference is gone.
And that's the brief explanation on the use of *may become* and not *becomes*.